anytree/exporter/mermaidexporter.py

```diff
@@ -0,0 +1,192 @@
+        >>> s0b = Node("sub0B", parent=s0, foo=4, edge=109)
+        >>> s0a = Node("sub0A", parent=s0, edge="")
+        >>> s1 = Node("sub1", parent=root, edge="")
+        >>> s1a = Node("sub1A", parent=s1, edge=7)
+        >>> s1b = Node("sub1B", parent=s1, edge=8)
+        >>> s1c = Node("sub1C", parent=s1, edge=22)
+        >>> s1ca = Node("sub1Ca", parent=s1c, edge=42)
+
+        A directed graph:
+
+        >>> from anytree.exporter import MermaidExporter
+        >>> for line in MermaidExporter(root):
+        ...     print(line)
+        graph TD
+        A[root] --> B[sub0]
+        A[root] --> C[sub1]
+        B[sub0] --> D[sub0B]
+        B[sub0] --> E[sub0A]
+        C[sub1] --> F[sub1A]
+        C[sub1] --> G[sub1B]
+        C[sub1] --> H[sub1C]
+        H[sub1C] --> I[sub1Ca]
+        """
+        self.node = node
+        self.graph = graph
+        self.name = name
+        self.options = options
+        self.indent = indent
+        self.nodenamefunc = nodenamefunc
+        self.nodeattrfunc = nodeattrfunc
+        self.edgeattrfunc = edgeattrfunc
+        self.edgetypefunc = edgetypefunc
+        self.maxlevel = maxlevel
+
+    def __iter__(self):
+        # prepare
+        indent = " " * self.indent
+        nodenamefunc = self.nodenamefunc or self._default_nodenamefunc
+        nodeattrfunc = self.nodeattrfunc or self._default_nodeattrfunc
+        edgeattrfunc = self.edgeattrfunc or self._default_edgeattrfunc
+        edgetypefunc = self.edgetypefunc or self._default_edgetypefunc
+        return self.__iter(indent, nodenamefunc, nodeattrfunc, edgeattrfunc, edgetypefunc)
+
+    @staticmethod
+    def _default_nodenamefunc(node):
+        return node.name
+
+    @staticmethod
+    def _default_nodeattrfunc(node):
+        # pylint: disable=W0613
+        return None
+
+    @staticmethod
+    def _default_edgeattrfunc(node, child):
+        # pylint: disable=W0613
+        return None
+
+    @staticmethod
+    def _default_edgetypefunc(node, child):
+        # pylint: disable=W0613
+        return "->"
+
+    def __iter(self, indent, nodenamefunc, nodeattrfunc, edgeattrfunc, edgetypefunc):
+        yield "{self.graph} {self.name}".format(self=self)
+        for option in self.__iter_options(indent):
+            yield option
+        for node in self.__iter_nodes(indent, nodenamefunc, nodeattrfunc):
+            yield node
+        yield ""
+
+    def __iter_options(self, indent):
+        options = self.options
+        if options:
+            for option in options:
+                yield "%s%s" % (indent, option)
+
+    def __iter_nodes(self, indent, nodenamefunc, nodeattrfunc):
+        def get_key(letter_index):
+            return chr(65 + letter_index)
+
+        index = 0
+        for node in PreOrderIter(self.node, maxlevel=self.maxlevel):
+            node.key = get_key(index)
+            index += 1
+
+        for node in PreOrderIter(self.node, maxlevel=self.maxlevel):
+            nodekey = node.key
```

Comment:
Each Node gets a unique capitalised letter assigned.

If you go to https://mermaid.live/ and generate the sample Flow diagram, it is like:

```
graph TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```

The capital letters are just node identifiers; the visual labels are in []/()/{}
Here is how this example renders:

```mermaid
graph TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```

Reply:
The referenced code has been updated to deduplicate iterations, but the key idea of this comment is the same.